distance-metrics/src/types.rs

```diff
@@ -0,0 +1,2 @@
+pub type VectorElementType = f32;
```

Comment:
In general, I don't think using type aliases for this kind of thing is very helpful in Rust code. I suppose this might have been useful for the comparisons with qdrant, but I'd prefer to inline this (note that Rust-Analyzer should be able to do this for you).